libs/sdk-core/src/breez_services.rs

```diff
@@ -1368,6 +1368,38 @@ impl BreezServices {
             },
         })
     }
+
+    pub async fn create_offer(&self, req: CreateOfferRequest) -> Result<String> {
+        self.start_node().await?;
+        Ok(
+            self
+                .node_api
+                .create_offer(req)
+                .await?
+        )
+    }
+
+    pub async fn pay_offer(&self, req: PayOfferRequest) -> Result<SendPaymentResponse, SendPaymentError> {
+        self.start_node().await?;
+
+        let fetch_invoice_response = self.node_api
+            .fetch_invoice(FetchInvoiceRequest {
+                offer: req.offer.clone(),
+                amount_msat: req.amount_msat,
+                timeout: req.timeout,
+                payer_note: req.payer_note.to_owned()
+            })
+            .await?;
+
+        if let Some(new_amount) = fetch_invoice_response.new_amount_msat {
+            return Err(SendPaymentError::OfferAmountChanged { new_amount });
+        }
```

Comment:
I wonder whether it could be useful to fail on all changes here. Given we don't support changes at all.

Reply:
Alright, I'll revert the changes to how they were previously and display all of them with an appropriate error